skimage/restoration/deconvolution.py

```diff
@@ -375,16 +376,27 @@ def richardson_lucy(image, psf, iterations=50, clip=True):
         convolve_method = fftconvolve
     else:
         convolve_method = convolve
-
     image = image.astype(np.float)
-    psf = psf.astype(np.float)
-    im_deconv = np.full(image.shape, 0.5)
+    psf = psf.astype(np.float)    
     psf_mirror = psf[::-1, ::-1]
-
-    for _ in range(iterations):
+    im_deconv = image
+    im_deconv_mid = np.zeros_like(image)
+    v = np.zeros_like(image)
+    for iter in range(iterations):
+        im_deconv_mid_prev = im_deconv_mid
         relative_blur = image / convolve_method(im_deconv, psf, 'same')
-        im_deconv *= convolve_method(relative_blur, psf_mirror, 'same')
-
+        im_deconv_mid = im_deconv * convolve_method(relative_blur, psf_mirror,
+                                                    'same')
+        v_prev = v
+        v = im_deconv_mid - im_deconv
+        np.maximum(v, 0, out=v)
+        if iter == 0:
+            alpha = 0
+        else:
+            alpha = (v_prev * v).sum() / ((v_prev * v_prev).sum() + 2e-15)
+            alpha = np.clip(alpha, 0, 1)
+        im_deconv = im_deconv_mid + alpha * (im_deconv_mid - im_deconv_mid_prev)
+        np.maximum(im_deconv, 0, out=im_deconv)
     if clip:
         im_deconv[im_deconv > 1] = 1
```

Comment:
I know that you did not wrote this, but `np.clip` is clearly advised here ;-)

Reply:
> Interesting, thank you for this contribution. I suppose that the acceleration is in term of convergence speed, so may be the default iterations count should be modified...

I'm not familiar with `pytest`. Do I have to pass the test?

Reply:
Please have a look at our contribution guide. The tests should definitely pass. That said, modifying them might be allowed in certain circumstances.

Reply:
Hi @WeisongZhao, test are supposed to pass for a PR to be merged.
You can run `pytest` locally
```
cd path_to_your_local_repository
pytest skimage/restoration
```
and if errors occurs, you can try to understand what happens using
```
pytest --pdb skimage/restoration
```
If you find a solution making the tests pass (bug fix in your code, values update in the tests...), you can implement it and submit it to review.
Remember what @hmaarrfk advised ;-)
> You want to be very careful changing the test. Tests should really change "as little as possible".